cluster-sharding/src/main/scala/com/evolutiongaming/akkaeffect/cluster/sharding/ClusterSharding.scala

```diff
@@ -44,11 +45,17 @@ trait ClusterSharding[F[_]] {
     extractEntityId: ShardRegion.ExtractEntityId,
     extractShardId: ShardRegion.ExtractShardId
   ): Resource[F, ActorRef]
+
+  /**
+   * Return IDs of local shards. Local shard is one assigned to locally managed shard region.
+   * Carefully: the implementation based on sending messages to all local shard regions and could be slow.
+   */
+  def localShards: F[List[String]]
```

Comment:
Should be `F[Map[String, List[String]]` or even better,
two separate methods to be added

```scala
def regions: Set[String]

def shards(region: String): Set[String]
```

Reply:
I would really like to understand what is going to be usage of that? 
Would we really need `regions` as public method? If yes -- than surely I may agree on what @t3hnar proposed.

Reply:
So as `akka-effect` is quite generic thing and further shutdown strategy logic gonna be implemented on the next layer I like Yaroslav's proposal about API.

Reply:
For now there're no plans of using `regions` separately. Also having two methods makes `ClusterShardingLocal` way more complicated. I've changed returned type to `F[Map[String, Set[String]]]`

Reply:
> Also having two methods makes ClusterShardingLocal way more complicated

I don't agree with this argument

Reply:
Would be nice to see any reasons of having two independent methods then.

My reason to avoid it: `ClusterShardingLocal` implementation is Actor based and represents region as actor.  Having `def regions: Set[String]` means that the `ActorRef` should be converted to string and then string to `ActorRef`. I don't see strong requirements of such API right now and prefer to avoid making any universal API that can satisfy all

Reply:
> Also having two methods makes ClusterShardingLocal way more complicated 

so, here is before:

```scala
def localShards: F[Map[String, Set[ShardState]]] =
               for {
                 regions <- withActorContext(_.children.filterNot(_.path.name.endsWith("Proxy")))
                 shards <- regions.toList.parTraverse { region =>
                   val ask = Ask.fromActorRef[F](region)
                   for {
                     r <- ask(RegionMsg.State, 1.second).flatten
                     s <- r.castM[F, List[ShardId]]
                   } yield region.path.name -> s.map(ShardState(_, Set.empty)).toSet
                 }
               } yield shards.toMap
```

here is after

```scala
def regions = {
withActorContext(_.children.filterNot(_.path.name.endsWith("Proxy")))
}

def shards(region: String) = {
for {
                 region <- withActorContext(_.children.find(_.path.name == region ))
                 shards <- region.foldMapM { region =>
                   val ask = Ask.fromActorRef[F](region)
                   for {
                     r <- ask(RegionMsg.State, 1.second).flatten
                     shards <- r.castM[F, List[ShardId]]
                   } yield shards.toSet
                 }
               } yield shards
}
```

It does not look like overcomplicated

Reply:
> My reason to avoid it: ClusterShardingLocal implementation is Actor based and represents region as actor. Having def regions: Set[String] means that the ActorRef should be converted to string and then string to ActorRef. I don't see strong requirements of such API right now and prefer to avoid making any universal API that can satisfy all

Isn't `F[Map[String, Set[String]]` a conversion of `ActorRef` to `String` as well?

Reply:
> Isn't F[Map[String, Set[String]] a conversion of ActorRef to String as well?

No, that's `typeName`: `akka.cluster.sharding.ClusterSharding#internalStart`

Reply:
sure, this is exactly `regionName = typeName `, am I missing something?

Reply:
basically 

```scala
def localShards: F[Map[TypeName, Set[Shards]]] 
```

vs

```scala

def regions: F[Set[TypeName]] 

def shards(typeName: TypeName): F[Set[Shards]]
```

or replace `Set[Shards]` with `ShardingState` if you will

Reply:
I cannot find any evidence that `TypeName` is `ActorRef#path` :(

> replace Set[Shards] with ShardingState if you will
good one, haven't seen `TypeName` class, will use it

Reply:
it is more
```scala
TypeName is ActorRef.path.name
```